### ISSUE TITLE
In Julia0.7 testing Gtk fails with "syntax: more arguments than types for ccall"

### ISSUE THREAD

**Issue by colbec** [JuliaGraphics/Gtk.jl#317]
While trying to use ImageView in Julia7, (but not Julia5.2 where all is good):

```
Julia Version 0.7.0-DEV.650
Commit 7affd0ced0 (2017-06-19 01:07 UTC)
Platform Info:
  OS: Linux (x86_64-suse-linux)
  CPU: Intel(R) Core(TM) i5-4460  CPU @ 3.20GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.9.1 (ORCJIT, haswell)
Environment:
```
`Pkg.test("Gtk")` fails with:

```
- many warnings suppressed -
WARNING: deprecated syntax "type" at /home/colin/.julia/v0.7/Gtk/test/../src/lists.jl:484.
Use "mutable struct" instead.
ERROR: LoadError: LoadError: LoadError: syntax: more arguments than types for ccall
Stacktrace:
 [1] include_from_node1(::Module, ::String) at ./loading.jl:549
 [2] include(::Module, ::String) at ./sysimg.jl:14
 [3] include(::String) at /home/colin/.julia/v0.7/Gtk/test/../src/Gtk.jl:3
 [4] include_from_node1(::Module, ::String) at ./loading.jl:549
 [5] include(::Module, ::String) at ./sysimg.jl:14
 [6] include(::String) at ./client.jl:376
 [7] include_from_node1(::Module, ::String) at ./loading.jl:549
 [8] include(::Module, ::String) at ./sysimg.jl:14
 [9] process_options(::Base.JLOptions) at ./client.jl:310
 [10] _start() at ./client.jl:378
while loading /home/colin/.julia/v0.7/Gtk/test/../src/lists.jl, in expression starting on line 44
while loading /home/colin/.julia/v0.7/Gtk/test/../src/Gtk.jl, in expression starting on line 58
while loading /home/colin/.julia/v0.7/Gtk/test/runtests.jl, in expression starting on line 1
=================================[ ERROR: Gtk ]=================================

failed process: Process(`/home/colin/Downloads/julia7/usr/bin/julia -Cnative -J/home/colin/Downloads/julia7/usr/lib/julia/sys.so --compile=yes --depwarn=yes --code-coverage=none --color=yes --compilecache=yes --check-bounds=yes --startup-file=yes /home/colin/.julia/v0.7/Gtk/test/runtests.jl`, ProcessExited(1)) [1]

================================================================================
ERROR: Gtk had test errors

```

**Comment by timholy** [comment]
I think almost everything is broken on 0.7. Can you checkout `release-0.6` instead?

**Comment by yuyichao** [comment]
This is a genuine Gtk.jl bug https://github.com/JuliaGraphics/Gtk.jl/blob/master/src/lists.jl#L50

**Comment by tknopp** [comment]
I am still on 0.6 an currently don't have time to do the port to 0.7.

**Comment by yuyichao** [comment]
This bug affects all versions of julia .....

**Comment by tknopp** [comment]
it seems the number of arguments is not correct.

**Comment by lobingera** [comment]
@yuyichao, i just ran Pkg.test("Gtk") (on 0.5.2 and 0.6-rc1) and don't get an error (like that). Do you think https://github.com/JuliaGraphics/Gtk.jl/blob/master/src/lists.jl#L50 isn't tested? How do you connect this to the above error message?

**Comment by yuyichao** [comment]
> isn't tested

Seems like it.

The error message `syntax: more arguments than types for ccall` should be pretty clear?

**Comment by tknopp** [comment]
There is a test for this and Julia < 0.7 seems to just ignored our false definition. The strict argument check of 0.7 is thus of course a good thing

**Comment by tknopp** [comment]
https://github.com/JuliaGraphics/Gtk.jl/blob/master/test/gui.jl#L595

**Comment by tknopp** [comment]
sorry forget what I said. wrong widget

**Comment by tknopp** [comment]
I cannot find a test. @yuyichao: might it be that the check is done on parse time? Then it would not need a test to trigger the error

**Comment by yuyichao** [comment]
> might it be that the check is done on parse time

You mean on master? Yes, that's why it triggers an error with a backtrace pointing to the loading of the file defining the function.

```
while loading /home/colin/.julia/v0.7/Gtk/test/../src/lists.jl, in expression starting on line 44
while loading /home/colin/.julia/v0.7/Gtk/test/../src/Gtk.jl, in expression starting on line 58
while loading /home/colin/.julia/v0.7/Gtk/test/runtests.jl, in expression starting on line 1
```

**Comment by tknopp** [comment]
fixed in #317
